### PR TITLE
Added linking user to external providers

### PR DESCRIPTION
**Pull request by agent3bood**

### CODE REVIEW

File: _includes/js/users.md [hunk 1]
```diff
@@ -395,3 +395,146 @@ The Facebook Javascript SDK provides a main `FB` object that is the starting poi
...
+
+## Linking Users
+
+Parse allows you to link your users with services like Twitter and Facebook, enabling your users to sign up or log into your application using their existing identities. This is accomplished through \_linkWith() method by providing authentication data for the service you wish to link to a user in the `authData` field. Once your user is associated with a service, the `authData` for the service will be stored with the user and is retrievable by logging in.
+
+`authData` is a JSON object with keys for each linked service containing the data below. In each case, you are responsible for completing the authentication flow (e.g. OAuth 1.0a using hellojs on client side) to obtain the information the the service requires for linking.
+
+### Facebook `authData`
+
+<pre><code class="json">
```
Comment (flovilmart) [inline _includes/js/users.md]: I believe we support the tripe back quotes.

**Review by flovilmart — COMMENTED** [review]
Thanks for the PR, can you try with the triple back quotes, I belive we support them:

\`\`\`json

\`\`\`

File: _includes/js/users.md [hunk 2]
```diff
@@ -395,3 +395,146 @@ The Facebook Javascript SDK provides a main `FB` object that is the starting poi
...
+### Unlinking
+
+Unlinking an existing user with a service also uses \_linkWith() method to clear `authData` from the user by setting the `authData` for the service to `null`.  For example, unlinking a user with a Facebook account would use a request like this:
+
+<pre><code class="javascript">
+let user = new Parse.User();
+user.id = "uMz0YZeAqc";
+user._linkWith("facebook", {}).then(function(user){
+  // user is unlinked now
+  });
```
Comment (flovilmart) [inline _includes/js/users.md]: remove space here for alignment

File: _includes/js/users.md [hunk 3]
```diff
@@ -395,3 +395,146 @@ The Facebook Javascript SDK provides a main `FB` object that is the starting poi
...
+let myAuthData = {
+  id: "123456789",
+  "access_token": "SaMpLeAAibS7Q55FSzcERWIEmzn6rosftAr7pmDME10008bWgyZAmv7mziwfacNOhWkgxDaBf8a2a2FCc9Hbk9wAsqLYZBLR995wxBvSGNoTrEaL",
+  "expiration_date": "2012-02-28T23:49:36.353Z"
+}
+let user = new Parse.User();
+user.id = "uMz0YZeAqc";
+user._linkWith("facebook", myAuthData).then(function(user){
+  // user is linked now
+  });
```
Comment (flovilmart) [inline _includes/js/users.md]: remove space here

File: _includes/js/users.md [hunk 4]
```diff
@@ -395,3 +395,146 @@ The Facebook Javascript SDK provides a main `FB` object that is the starting poi
...
+  "screen_name": "ParseIt",
+  "consumer_key": "SaMpLeId3X7eLjjLgWEw",
+  "consumer_secret": "SaMpLew55QbMR0vTdtOACfPXa5UdO2THX1JrxZ9s3c",
+  "auth_token": "12345678-SaMpLeTuo3m2avZxh5cjJmIrAfx4ZYyamdofM7IjU",
+  "auth_token_secret": "SaMpLeEb13SpRzQ4DAIzutEkCE2LBIm2ZQDsP3WUU"
+}
+let user = new Parse.User();
+user._linkWith('twitter', myAuthData).then(function(user){
+    // user
+  });
```
Comment (flovilmart) [inline _includes/js/users.md]: remove space here

**Review by flovilmart — CHANGES_REQUESTED** [review]
also a few code nits ;)

File: _includes/js/users.md (hunk 1 above)
Comment (agent3bood) [reply _includes/js/users.md]: changed the class to javascript

**Comment by agent3bood** [comment]
done, please check and tell me if any this is wrong

**Comment by flovilmart** [comment]
This is looking great! Thanks!!!

**Review by flovilmart — APPROVED** [review]

**Pull request merged**